### PR TITLE
apidump: Dump functions before callind ICD

### PR DESCRIPTION
**Pull request by charles-lunarg**
Functions without a result or a non-const pointer
parameter now will dump their values before calling
into the ICD. Calls that crash the driver can be
dumped before the crash happens, helping to diagnose
bad calls.

files modified:
	scripts/api_dump_generator.py

Change-Id: I4b8ecbcb43875055f7a5cc6253316be00ba963c0

### CODE REVIEW

File: scripts/api_dump_generator.py [hunk 1]
```diff
@@ -434,6 +446,8 @@
 std::ostream& dump_text_{unName}(const {unName}& object, const ApiDumpSettings& settings, int indents);
 @end union
 
+//======================== pNext Chain Implementation =======================//
+
```
Comment (davidlunarg) [inline scripts/api_dump_generator.py]: Best to not include changes not related to what this pull request implements/fixes. Submit a separate pull request.

**Comment by charles-lunarg** [comment]
What cases are there were parameters are modified? 
Currently any non const pointer parameters disable the early dumping. Are there places where parameters are modfied other than this?

**Comment by charles-lunarg** [comment]
Fixed the vktracedump issue that the previous commit worked around. Compile time should be more reasonable now.
